### PR TITLE
[MISC] Rename core/concept/tuple.hpp -> utility/tuple/concept.hpp

### PR DESCRIPTION
**Pull request by simonsasse**
- [x] concept/cereal.hpp -> core/concept/cereal.hpp (wasn't much to do here 🥳)
- [ ] concept/core_language.hpp -> utility/concept/.  ? Are all concepts relevant - maybe exposition only concepts should not be exposed! (postponed until a decision is made )
- [x] concept/tuple.hpp ->utility -> utility/tuple/concept.hpp

### CODE REVIEW

File: include/seqan3/core/concept/all.hpp [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 #include <seqan3/core/concept/cereal.hpp>
 #include <seqan3/core/concept/core_language.hpp>
-#include <seqan3/core/concept/tuple.hpp>
+#include <seqan3/utility/tuple/concept.hpp> //can be removed without test failure.
```
Comment (marehr) [inline include/seqan3/core/concept/all.hpp:17]:
```suggestion
#include <seqan3/utility/tuple/concept.hpp>
```

true, but `all.hpp` should include all headers

File: include/seqan3/core/concept/tuple.hpp [hunk 2]
```diff
@@ -6,203 +6,14 @@
 // -----------------------------------------------------------------------------------------------------
 
 /*!\file
- * \brief Provides seqan3::tuple_like.
+ * \brief [DEPRECATED] Provides seqan3::tuple_like.
  * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ * \deprecated This header will be removed in 3.1. Please use seqan3/utility/tuple/concept.hpp instead.
```
Comment (marehr) [inline include/seqan3/core/concept/tuple.hpp:11]:
```suggestion
 * \deprecated This header will be removed in SeqAn-3.1.0. Please use seqan3/utility/tuple/concept.hpp instead.
```

File: test/unit/alignment/pairwise/align_result_selector_test.cpp [hunk 3]
```diff
@@ -21,7 +21,7 @@
 #include <seqan3/alignment/pairwise/align_result_selector.hpp>
 #include <seqan3/alphabet/gap/gapped.hpp>
 #include <seqan3/alphabet/nucleotide/dna4.hpp>
-#include <seqan3/core/concept/tuple.hpp>
+#include <seqan3/utility/tuple/concept.hpp>
 #include <seqan3/range/views/type_reduce.hpp>
```
Comment (marehr) [inline test/unit/alignment/pairwise/align_result_selector_test.cpp:25]:
```suggestion
#include <seqan3/range/views/type_reduce.hpp>
#include <seqan3/utility/tuple/concept.hpp>
```

File: include/seqan3/utility/tuple/all.hpp [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+#include <seqan3/utility/tuple/concept.hpp>
+
+/*!\defgroup concept Concept
+ * \ingroup utility
+ * \brief Additional concepts that are not specific to a SeqAn module.
+ *
+ * \details
+ *
+ * \todo write me.
+ */
```
Comment (marehr) [inline include/seqan3/utility/tuple/all.hpp:24]:
```suggestion
/*!\defgroup tuple Tuple
 * \ingroup utility
 * \brief Additional helper utilities for "tuple" types like std::tuple, std::pair, seqan3::pod_tuple that are not specific to a SeqAn module.
 *
 * \details
 *
 * \todo write me.
 */
```

**Review by marehr — CHANGES_REQUESTED** [review]
Thank you!

File: include/seqan3/core/concept/all.hpp (hunk 1 above)
Comment (simonsasse) [reply include/seqan3/core/concept/all.hpp:17]: ok :)

File: include/seqan3/core/concept/tuple.hpp (hunk 2 above)
Comment (simonsasse) [reply include/seqan3/core/concept/tuple.hpp:11]: Thanks !

File: test/unit/alignment/pairwise/align_result_selector_test.cpp (hunk 3 above)
Comment (simonsasse) [reply test/unit/alignment/pairwise/align_result_selector_test.cpp:25]: 👌

**Comment by simonsasse** [comment]
rebased and changes applied :)

**Comment by simonsasse** [comment]
New merge conflicts. I'll rebase in the next hours.

**Comment by simonsasse** [comment]
Rebased.

File: include/seqan3/alignment/pairwise/detail/policy_optimum_tracker.hpp [hunk 5]
```diff
@@ -19,8 +19,9 @@
 #include <seqan3/alignment/matrix/detail/matrix_coordinate.hpp>
 #include <seqan3/alignment/pairwise/detail/type_traits.hpp>
 #include <seqan3/core/algorithm/configuration.hpp>
-#include <seqan3/core/concept/tuple.hpp>
 #include <seqan3/core/detail/template_inspection.hpp>
+#include <seqan3/core/detail/template_inspection.hpp>
```
Comment (marehr) [inline include/seqan3/alignment/pairwise/detail/policy_optimum_tracker.hpp:23]:
```suggestion
```

**Review by marehr — APPROVED** [review]
lgtm; remove that one double include :)

File: include/seqan3/alignment/pairwise/detail/policy_optimum_tracker.hpp (hunk 5 above)
Comment (simonsasse) [reply include/seqan3/alignment/pairwise/detail/policy_optimum_tracker.hpp]: whoops. Done ! Thx

File: include/seqan3/io/alignment_file/detail.hpp [hunk 6]
```diff
@@ -20,11 +20,12 @@
 
 #include <seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp>
 #include <seqan3/alphabet/cigar/cigar.hpp>
-#include <seqan3/core/concept/tuple.hpp>
+
```
Comment (smehringer) [inline include/seqan3/io/alignment_file/detail.hpp:23]: remove the new line here

File: include/seqan3/io/alignment_file/format_sam.hpp [hunk 7]
```diff
@@ -20,7 +20,7 @@
 #include <vector>
 
 #include <seqan3/core/concept/core_language.hpp>
-#include <seqan3/core/concept/tuple.hpp>
+
```
Comment (smehringer) [inline include/seqan3/io/alignment_file/format_sam.hpp:23]: and here

File: include/seqan3/io/detail/record.hpp [hunk 8]
```diff
@@ -13,12 +13,11 @@
  */
 
 #pragma once
-
```
Comment (smehringer) [inline include/seqan3/io/detail/record.hpp:16]: here you removed the new but shouldn't :)

File: test/include/seqan3/test/performance/units.hpp [hunk 9]
```diff
@@ -12,14 +12,16 @@
 
 #pragma once
 
+#include <seqan3/std/ranges>
+
```
Comment (smehringer) [inline test/include/seqan3/test/performance/units.hpp:16]: there already is a std/ranges include below the benchmark include? I think you can delete this one again

**Review by smehringer — CHANGES_REQUESTED** [review]
Just a few things that got mixed up in between :)

**Comment by simonsasse** [comment]
Review applied :)

**Review by smehringer — APPROVED** [review]

**Comment by smehringer** [comment]
@eaasna This branch needs a rebase on to the current master. (`git remote update`, `git rebase upstream/master`). And then resolve the merge conflicts. If you have troubles we can also trouble shoot on Wednesday

**Pull request merged**